File: python/misc/sets.py
# sets are unordered collections
my_set = {1.0, 10, "String", (1, 0, 1, 0)}

print(my_set)
# {'String', 1.0, 10, (1, 0, 1, 0)}
print("String" in my_set)
# True

# add
my_set.add("Python")
print(my_set)
# {1.0, 'String', 10, (1, 0, 1, 0), 'Python'}

# remove
my_set.remove("String")
print(my_set)
# {1.0, 10, (1, 0, 1, 0), 'Python'}

# set operations
set_one = {1, 2, 3}
set_two = {3, 4, 5}

# union
print(set_one | set_two)
# {1, 2, 3, 4, 5}

# Intersection
print(set_one & set_two)
# {3}

# Difference
print(set_one - set_two)
# {1, 2}

# Symmetric difference
print(set_one ^ set_two)
# {1, 2, 4, 5}

# subset and superset
set_a = {1, 2}
set_b = {1, 2}
set_c = {1, 2, 3, 4, 5}

# Strict subset
print(set_a < set_b)
# False

# Subset
print(set_a <= set_b)
# True

File: _Pascal/Primer/sets.pas
// 2020-02
program Sets;

type
    // sets are collections of same type
    days_of_week = set of (Mon, Tue, Wed, Thu, Fri, Sat, Sun);
    letters = set of char;
    alphabet = set of 'A' .. 'Z';
var
    // declare sets
    A: alphabet;
    B: alphabet;
    C: alphabet;

    days: days_of_week;

    // declare iterator variables
    i: char;
begin
    A := ['A', 'B', 'C', 'D'];
    for i in A do write(i, ' '); writeln();
    // A B C D

    B := ['C', 'D', 'E', 'F'];
    for i in B do write(i, ' '); writeln();
    // C D E F

    // union
    for i in (A + B) do write(i, ' '); writeln();
    // A B C D E F 

    // intersection
    for i in (A * B) do write(i, ' '); writeln();
    // C D 

    // difference
    for i in (A - B) do write(i, ' '); writeln();
    // A B

    // symmetric difference
    for i in (A >< B) do write(i, ' '); writeln();
    // A B E F 

    // subset
    C := ['A', 'B', 'C'];
    if (C <= A) then writeln(true);
    // TRUE

    // include
    include(A, 'E');
    for i in A do write(i, ' '); writeln();
    // A B C D E

    // exclude
    exclude(A, 'E');
    for i in A do write(i, ' '); writeln();
    // A B C D

    // in
    if 'A' in A then writeln(true);
    // TRUE

    days := [Mon, Wed, Fri];
    if Wed in days then writeln(true);
    // TRUE
end.